### ISSUE TITLE
Inquiry Regarding IAMDataset Preprocessing

### ISSUE THREAD

**Issue by Kimchangheon** [GemCarr/handwriting_reco#1]
Hello! I hope this message finds you well.
I'm Changhun from FAU Erlangen. I'm utilizing your code for my project with Dr. Vincent Christlein. 

I'm deeply impressed by your concise and clear code structure; it's going to be very helpful to me. However, to utilize your code, I'm wondering if I can access the dataset you used in this repo.

I'd like to know how you preprocessed the IAM dataset. What I know is the original IAM dataset needs to be preprocessed into images and their corresponding text, as shown below.

a01-000u-00.png
![a01-000u-00](https://github.com/GemCarr/handwriting_reco/assets/39324421/7af535eb-f90e-47b6-907a-767be03a1f0f)

.txt
a01-000u-00.png	A MOVE to stop Mr. Gaitskell from

Could you kindly explain how you preprocessed the dataset, or provide the preprocessed dataset directly?

**Comment by GemCarr** [comment]
Hello,
I have to check in order to answer your questions about the dataset, which I cannot do at the moment. Before monday I should be able to answer and give you the preprocessing code and links to the dataset.

Best regards,
Guillaume Carriere

------- Original Message -------

**Comment by GemCarr** [comment]
I added the .json file and instructions for constructing the data.
Adding the corresponding text from the images implies going through the xml data given on the IAM website, to save you the hassle of doing that I can personally send you the processed data privately if you want

**Comment by Kimchangheon** [comment]
Dear Guillaume Carriere,

Thank you very much for your response. I have followed a similar process, and I hope it's correct. https://drive.google.com/drive/folders/1TVTPmsBl9aWGCAzRCTz_jJeyQFFKqIJ1?usp=sharing
BTW, I'm eager to access the processed data. Could you please send it to [changhun.kim@fau.de](mailto:changhun.kim@fau.de)?

Additionally, I'm unsure about how to use the following package:
`from augments.ocrodeg import OcrodegAug`
It seems to be a user-defined package. Could I also gain access to that?

Furthermore, when running the training, I encountered an error. Although I'm trying to debug it, I would greatly appreciate your expert opinion on the cause of this issue.

Best regards,
Changhun Kim

<img width="1149" alt="image" src="https://github.com/GemCarr/handwriting_reco/assets/39324421/be3f6559-f1c4-4adf-8254-8d690ae3391b">

**Comment by GemCarr** [comment]
From what I can see in your drive it looks like the images are missing, but I will send you the data anyways directly at your address.

The OcrodegAug class in defined in this file https://github.com/GemCarr/handwriting_reco/blob/main/augments/ocrodeg.py on the repository.

On the error, it may be due to a difference in the data you are using, or maybe due to the transforms. If you still encounter the error with the data I will provide I'll look into it.

**Comment by Kimchangheon** [comment]
Thank you very much for your reply!
Oh It's my bad. Acutally, I wanted to ask about importing this in ocrodeg.py:
`from htr.data.augmentation import ocrodeg_utils`

**Comment by GemCarr** [comment]
Ah true it looks like its missing, will upload it asap

**Comment by Kimchangheon** [comment]
If you don't mind, could you kindly explain the following this importing?
from htr.data.augmentation import ocrodeg_utils.

I guess it's user-defined. Could I also obtain this module as well?

**Comment by GemCarr** [comment]
Yes that what I meant by uploading, i forgot to include the corresponding file where these functions are defined
